frontend/scripts/metrica.js

```diff
@@ -0,0 +1,309 @@
+(() => {
+    class requestService {
+        post(url, data, headers = {}) {
+            return fetch(url, {
+                method: 'POST',
+                crossDomain: true,
+                headers: headers,
+                body: data
+            });
+        }
+    }
+    const state = {
+        host: 'http://stage.metrica.fun/api/v1',
+        routes: {
+            create_visitor: '/visitors',
+            create_visit: '/visits'
+        }
+    };
+    window._metricaTracking = {
+        endTime:undefined,
+        configMetrica:undefined,
+        initialize(object, prop) {
+            this.setObjectMetricaConf(object, prop);
+            let token = this.getToken();
+            if (token) {
+                this.createVisit();
+            } else {
+                this.createVisitor(this.getTrackById())
+                    .finally(() => this.createVisit());
+            }
+
+        },
+        setObjectMetricaConf(object, prop) {
+            if(Helper.isObjectMetricaConfig(object, prop)) {
```

Comment:
better take explicitly _metricaTrackingConfig from "window" object. And if it is an array then check if it has startDate property and tracking number. If not then it means your script has loaded earlier then script on the page was executed. It is possible when script was retrieved from browser cache. In this case you should fill config yourself, retrieving tracking number from script tag.